// Copyright (c) 2020-2022 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.protobuf;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTBlockchainEthereumAccountData";
option optimize_for = LITE_RUNTIME;

import public "BlockchainImportedAccountData.proto";
import public "HDPath.proto";

message BlockchainEthereumAccountData
{
    optional uint32 version = 1;
    optional BlockchainImportedAccountData imported = 2;
    optional bytes balance = 3;
    optional int64 next_nonce = 4;
    repeated int64 known_outgoing = 5;
    repeated bytes incoming_txid = 6;
    repeated bytes outgoing_txid = 7;
    optional HDPath path = 8;
    optional uint32 standard = 9;
}
